verifier/verifier.go

```diff
@@ -412,6 +420,20 @@ func verifyAuthenticity(ctx context.Context, trustPolicy *trustpolicy.TrustPolic
 	}
 }
 
+func verifyUserMetadata(logger log.Logger, payload *envelope.Payload, userMetadata map[string]string) error {
+	logger.Debugf("Verifying that metadata %v is present in signature", userMetadata)
+	logger.Debugf("Signature metadata: %v", payload.TargetArtifact.Annotations)
+
+	for k, v := range userMetadata {
+		if payload.TargetArtifact.Annotations[k] != v {
```

Comment:
In case of an edge case, users may require a field to be present but empty.
```suggestion
		if got, ok := payload.TargetArtifact.Annotations[k]; !ok || got != v {
```